lib/practice_exercises.rb

```diff
@@ -1,13 +1,36 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  words = ""
+  string.downcase.each_char do |char|
+    if char.match? /[0-9a-z]/
+      words += char
+    end
+  end
+
+  words.each_char.each_with_index do |char, i|
+    unless words[-1-i] == char
+      return false
+    end
+  end
+  true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
Clever.  However this method Find the greatest and smallest strings O(n) and then loops through the smallest string trying to find where they stop matching O(m).  So this solution is O(n + m) in time complexity.  

On the other hand you are getting a string and building a prefix so this would be O(m) space complexity.

Reply:
Thank you Chris, After the lecture, I'm understanding more about the O(m)